examples/nffPktgen/testing/minPktLoss.go

```diff
@@ -0,0 +1,214 @@
+                low := 0
+                high := 100
+                currSpeed := 100
+
+                // data at line rate:
+                portStats.packetCount = 0
+                gen.Count = 0
+                time.Sleep(time.Duration(TRAFFIC_DELAY) * time.Second)
+                pktRX := portStats.packetCount
+                pktTX := gen.GetGeneratedNumber()
+                pktLoss := calcPktLoss(pktRX, pktTX)
+                totalPktRX += pktRX
+                totalPktTX += pktTX
+                printStats(&portStats, pktRX, pktTX, pktLoss, started, currSpeed, float64(TOP_SPEED))
+
+                // binary search:
+                for low <= high {
+                        mid := (low + high) / 2
+                        currSpeed = mid
+                        updatedSpeed := 0.01 * float64(uint64(currSpeed) * TOP_SPEED)
+
+                        // reset counters:
+                        portStats.packetCount = 0
+                        gen.Count = 0
+
+                        genChan <- uint64(updatedSpeed)
+                        time.Sleep(time.Duration(TRAFFIC_DELAY) * time.Second)
+
+                        pktRX = portStats.packetCount
+                        pktTX = gen.GetGeneratedNumber()
+                        pktLoss = calcPktLoss(pktRX, pktTX)
+
+                        tgt := pktLoss <= TGT_LOSS
+                        if(tgt) {
+                                low = mid + 1 // tgt met so try higher speed
+                                if currSpeed > maxSpeed {
+                                        maxSpeed = currSpeed
+                                }
+                        } else {
+                                high = mid - 1 // tgt failed so try lower speed
+                        }
+
+                        totalPktRX += pktRX
+                        totalPktTX += pktTX
+                        //fmt.Printf("\nmid: %d\nupdatedSpeed: %d%%(%f pkt/s)\npktCountTX: %d\npktCountRX: %d\npktLoss: %f\nlow: %d\nhigh: %d\n\n", mid, currSpeed, updatedSpeed, pktTX, pktRX, pktLoss, low, high) // debugging purposes
+                        printStats(&portStats, pktRX, pktTX, pktLoss, started, currSpeed, updatedSpeed)
+                }
+                fmt.Printf("-----------------------------------------------------------BINARY SEARCH COMPLETE!!!------------------------------------------------------------\n")
+                interruptChannel <- os.Interrupt
+        }()
+
+
+out:
+        for {
+                select {
+                case sig := <-interruptChannel:
+                        fmt.Printf("Received signal %v, finishing.\n", sig)
+                        break out
+                case <-finishChannel:
+                        fmt.Println("Test timeout reached")
+                        break out
+                case <-statsChannel:
+                        //printStats(&portStats, totalPktRX, totalPktTX, calcPktLoss(totalPktRX, totalPktTX), started, maxSpeed, /*pkt/s*/)
```

Comment:
I think you can uncomment this line because statistics are disabled by default in parameters, but may be needed in some cases.